h2/src/main/org/h2/mvstore/Page.java

```diff
@@ -947,10 +808,24 @@ public void removePage() {
         }
     }
 
+    public abstract void removeAllRecursive();
+
+    private Object[] createKeyStorage(int size)
+    {
+        return new Object[size];
+    }
+
+    protected final Object[] createValueStorage(int size)
+    {
+        return new Object[size];
+    }
+
```

Comment:
these create methods seem too simple to warrant existing?

Reply:
Well, than they will be in-lined anyway. I am going back from my version where keys/values may be native, and then those methods are less trivial. So just to keep track of dozen or so call sites...

Reply:
fair enough